### DISCUSSION TITLE
Feature request: Interactive script details also for SQL query strings

### DISCUSSION THREAD

**Discussion by eduardGradenwitz** [burkybang/SuiteAdvanced-Feedback#207] [Ideas]
like a copy button or even a button to run with suiteQL 
During debugging i often log the query first and then tweak it in your run quiteQL page

**Comment by burkybang** [comment]
Interesting. The difficult part would be coming up with a way to programmatically detect SuiteQL.

**Reply by eduardGradenwitz** [reply]
So you think Starting with uppercase SELECT would be to lazy. 
Hm

**Reply by burkybang** [reply]
There could be too many false positives. Some people don't use uppercase keywords, and SuiteQL doesn't have to start with SELECT.

**Reply by eduardGradenwitz** [reply]
Maybe require a pre tag
Like 
--#suiteadvanced
SELECT [...]

That's confusing to all non SA developer's though.

**Reply by eduardGradenwitz** [reply]
Or better
Title= query or contains query
And details contains select in the first 100 chars or first 2 lines
Or starts with select
That's not to bad and similar to the record links

**Reply by eduardGradenwitz** [reply]
Record links also enforce a syntax. so requiring something is not new.
You got to have something. Also you can not do a huge regex since that costs something I guess.

**Reply by eduardGradenwitz** [reply]
On that note for record links

Now This works
{
salesorder : 124,


//What also would be cool is

salesOrderId : 123,

}
But that's a bit extra.
